src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/OpenDocumentGenerator.cs

```diff
@@ -167,17 +177,17 @@ private void ProjectManager_Changed(object? sender, ProjectChangeEventArgs args)
                 }
         }
 
-        void EnqueueIfNecessary(DocumentSnapshot document)
+        void EnqueueIfNecessary(DocumentKey documentKey, int documentVersion)
         {
-            if (!_projectManager.IsDocumentOpen(document.FilePath) &&
-                !_options.UpdateBuffersForClosedDocuments)
+            if (!_options.UpdateBuffersForClosedDocuments &&
+                !_projectManager.IsDocumentOpen(documentKey.FilePath))
             {
                 return;
             }
 
-            _logger.LogDebug($"Enqueuing generation of {document.FilePath} in {document.Project.Key.Id} at version {document.Version}");
+            _logger.LogDebug($"Enqueuing generation of {documentKey.FilePath} in {documentKey.ProjectKey.Id} at version {documentVersion}");
```

Comment:
Seems like this log message should be moved into the actual processing method, as the version of the the document being processed could be different to what is logged here. Alternatively, could leave this and add a new log in the process method with the version actually being dealt with, so we can tell if things are getting delayed

Reply:
Absolutely agreed.